src/components/CmsZone.vue

```diff
@@ -225,25 +235,15 @@ export default class CmsZone extends Vue {
       ? `<div class="zone-footer">${data.zone_footer || ''}</div>`
       : '';
     this.zoneStatus = null;
+
     // Circumvent issue where carousel breaks by forcing it to re-render
     this.nonce++;
+    this.setupTracking(this.contents);
+  }
 
-    if (!this.contents.length) {
-      return;
-    }
-
-    this.removeScrollListeners();
-    this.scrollable = getClosest(this.$el, '.scrollable-content');
-    if (this.scrollable) {
-      this.trackScrollable(this.contents, this.scrollable);
-      return;
-    }
-
-    if (this.zoneType === 'carousel') {
-      this.trackIndex(0);
-    } else {
-      this.contents.forEach((c, i): void => this.trackIndex(i));
-    }
+  private isAlmostVisible() {
+    const elRect = this.$el.getBoundingClientRect();
+    return elRect.top < 1.25 * window.innerHeight;
```

Comment:
should we also check if the bottom is visible in the off chance a users starts halfway down a page?

Reply:
yeah, would love to chat more about other things we should account for. There's also a separate problems where all zones being empty at the outset make it so that nothing lazy loads. Would love your help thinking through that.